named_curve.go

```diff
@@ -46,6 +48,13 @@ func generateKeypair(c namedCurve) (*namedCurveKeypair, error) {
 		}
 
 		return &namedCurveKeypair{namedCurveP256, elliptic.Marshal(elliptic.P256(), x, y), privateKey}, nil
+	case namedCurveP384:
+		privateKey, x, y, err := elliptic.GenerateKey(elliptic.P384(), rand.Reader)
+		if err != nil {
+			return nil, err
+		}
+
+		return &namedCurveKeypair{namedCurveP384, elliptic.Marshal(elliptic.P384(), x, y), privateKey}, nil
```

Comment:
Same here. If not possible totally understand! Just worried if someone find a bugs in one of these duplicated cases they won't fix the other

Reply:
Okay I combined them here a bit more. LMK if you want it even more abstracted